src/librbd/object_map/RefreshRequest.cc

```diff
@@ -133,6 +134,15 @@ Context *RefreshRequest<I>::handle_load(int *ret_val) {
     return nullptr;
   } else if (*ret_val < 0) {
     lderr(cct) << "failed to load object map: " << oid << dendl;
+    m_image_ctx.owner_lock.lock_shared();
+    if (*ret_val == -ETIMEDOUT &&
+        !m_image_ctx.get_exclusive_lock_policy()->may_auto_request_lock() &&
+        cct->_conf.get_val<bool>("rbd_auto_exclusive_lock_until_manual_request")) {
```

Comment:
I'd rather see a new developer-level configuration option to enable this behavior (since it's really corner-case). I would also like to see the invalidate issue fixed as well so that the `-ETIMEDOUT` error propagates back in the normal case.